src/components/navigation/SlideMenuItems.tsx

```diff
@@ -1,86 +1,123 @@
 import { ExpandLess, ExpandMore } from '@material-ui/icons';
 import { useHistory } from 'react-router-dom';
 
-import { NavigationItem, UtilityItem } from '../../types';
+import { NavigationItem, UtilityItem } from '@app/types';
 
-import { navigationItemParentMap } from './NavigationItems';
+import { useModal } from '@app/contexts/ui/ModalProvider';
 
-import useStyles from './SlideMenu.jss';
+import { useSlideMenuState } from './SlideMenuContext';
 
-const menuState = {
-  expandedNavItems: new Set<string>(),
-};
+import useStyles from './SlideMenu.jss';
 
 /**
  * List item for the navigation list in the slide menu
  */
 interface NavListItemProps extends NavigationItem {
-  menuClosingAction: () => void;
+  onRequestOpen: () => void;
+  onRequestClose: () => void;
+  isDrawerOpen: boolean;
 }
+
 const NavListItem: React.FC<NavListItemProps> = props => {
-  const [open, setOpen] = useState<boolean>(menuState.expandedNavItems.has(props.id));
+  const {
+    id,
+    subItemList,
+    path,
+    onRequestClose,
+    onRequestOpen,
+    isDrawerOpen,
+    isVisible,
+    icon,
+    text,
+  } = props;
   const history = useHistory();
+  const menuState = useSlideMenuState();
   const classes = useStyles();
-  const hasSubList = !!props.subItemList?.length;
-  const isSelected: boolean = history.location.pathname === props.path;
-
-  // @ATTENTION I'd like to use rrd's useHistory hook - will this cause any issues?
-  useEffect(() => {
-    history.listen(location => {
-      const activeItem = Object.values(navigationItemParentMap).find(
-        item => item.path === location.pathname,
-      );
-      if (activeItem) {
-        activeItem.parents.forEach(parent => menuState.expandedNavItems.add(parent));
-      }
-    });
-  }, [history]);
+  const hasSubList = !!subItemList?.length;
+  const sublistOpen = !!menuState?.expandedNavItems.includes(id);
+  const isSelected: boolean = ((): boolean => {
+    if (menuState?.activeNavItem && menuState?.activeNavItem?.path === path) return true;
+    if (isDrawerOpen) return false;
+    if (menuState?.activeNavItem?.parents.includes(id)) return true;
+    return false;
+  })();
 
-  const handleClick = (): void => {
-    if (hasSubList) {
-      if (open) {
-        menuState.expandedNavItems.delete(props.id);
-        setOpen(false);
+  const handleClick = useCallback((): void => {
+    if (!isDrawerOpen) {
+      onRequestOpen();
+    } else if (hasSubList) {
+      if (sublistOpen) {
+        menuState?.collapseNavItem(id);
       } else {
-        menuState.expandedNavItems.add(props.id);
-        setOpen(true);
+        menuState?.expandNavItem(id);
       }
-    } else if (props.path) {
-      history.push(props.path);
-      props.menuClosingAction();
+    } else if (path) {
+      history.push(path);
+      onRequestClose();
     }
-  };
+  }, [
+    hasSubList,
+    path,
+    isDrawerOpen,
+    sublistOpen,
+    onRequestOpen,
+    menuState,
+    id,
+    history,
+    onRequestClose,
+  ]);
```

Comment:
too many dependency array items to justify a useCallback - the only props this doesn't use are ones I would expect to never change, this isn't worth sacrificing legibility

Reply:
Yeah, I didn't notice that. Un-memoized https://github.com/resident51/resident51/pull/3/commits/ad9cf714da9c80a0a85a88d5891d7dee55f45ffa